### PR TITLE
Add test to catch regressions in sni and override configs

### PR DESCRIPTION
**Pull request by shinrich**
Created this to test a fix we made internally.  Turns out this fix was already made in the open source, but adding the test to catch potential future regressions seems like a good idea.

### CODE REVIEW

File: tests/gold_tests/tls/tls_verify_override_sni.test.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import os
+Test.Summary = '''
+Test tls server certificate verification options. Exercise conf_remap
+'''
+
+# need Curl
```
Comment (randall) [inline tests/gold_tests/tls/tls_verify_override_sni.test.py]: It was decided somewhere that we would assume that curl is on the host.

**Comment by shinrich** [comment]
Pushed a new version without the curl check.

**Comment by randall** [comment]
[approve ci freebsd]

**Comment by zwoop** [comment]
Cherry-picked to v9.0.x branch.

**Pull request merged**